services/121-service/src/migration/1764944177810-split-voucher-permissions.ts

```diff
@@ -0,0 +1,94 @@
+import { EntityManager, Equal, MigrationInterface, QueryRunner } from 'typeorm';
+
+import { PermissionEntity } from '@121-service/src/user/entities/permissions.entity';
+import { UserRoleEntity } from '@121-service/src/user/entities/user-role.entity';
+import { PermissionEnum } from '@121-service/src/user/enum/permission.enum';
+
+export class SplitVoucherPermissions1764944177810
+  implements MigrationInterface
+{
+  name = 'SplitVoucherPermissions1764944177810';
+  closestPermissionName = 'payment:voucher.read';
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.startTransaction();
+
+    // Add the new paper permission
+    await this.addNewPermission({
+      manager: queryRunner.manager,
+      newPermissionName: 'payment:voucher-paper.read',
+    });
+
+    // Add the new paper permission
+    await this.addNewPermission({
+      manager: queryRunner.manager,
+      newPermissionName: 'payment:voucher-whatsapp.read',
+    });
+
+    await queryRunner.commitTransaction();
+
+    // Get old permission ID and delete it from roles and permissions
+    const closestPermissionIdQuery: { id; number }[] = await queryRunner.query(`
```

Comment:
```suggestion
    const closestPermissionIdQuery: { id: number }[] = await queryRunner.query(`
```
this should be a colon right? But you've run this correctly right..?

Reply:
yes it works